pkg/platform/kube/functionres/lazy.go

```diff
@@ -1163,9 +1158,23 @@ func (lc *lazyClient) createOrUpdateCronJob(functionLabels labels.Set,
 	cronJobSpec.Suspend = &suspendCronJob
 
 	getCronJob := func() (interface{}, error) {
-		return lc.kubeClientSet.BatchV1beta1().
+		cronJobs, err := lc.kubeClientSet.BatchV1beta1().
 			CronJobs(function.Namespace).
-			Get(kube.CronJobNameFromFunctionName(function.Name, jobName), metav1.GetOptions{})
+			List(metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("nuclio.io/function-cron-trigger-cron-job=true,"+
+					"nuclio.io/function-name=%s,"+
+					"nuclio.io/function-cron-trigger-name=%s",
+					function.Name,
+					jobName),
+			})
+		if err != nil {
+			return nil, err
+		}
+		if len(cronJobs.Items) == 0 {
+			return nil, apierrors.NewNotFound(schema.GroupResource{},
```

Comment:
You can use nuclio sdk's `errors.NewNotFound` instead of k8s one (which is being used to identify k8s _real_ not found errors